File: app/src/main/java/com/novel/read/utlis/StringUtils.java
package com.novel.read.utlis;

import android.content.Context;
import android.text.TextUtils;

import com.spreada.utils.chinese.ZHConverter;

import static com.novel.read.widget.page.ReadSettingManager.SHARED_READ_CONVERT_TYPE;

public class StringUtils {

    /**
     * 将文本中的半角字符，转换成全角字符
     *
     * @param input
     * @return
     */
    public static String halfToFull(String input) {
        input = deleteImgs(input);
        char[] c = input.toCharArray();
        for (int i = 0; i < c.length; i++) {
            if (c[i] == 32) //半角空格
            {
                c[i] = (char) 12288;
                continue;
            }
            //根据实际情况，过滤不需要转换的符号
            //if (c[i] == 46) //半角点号，不转换
            // continue;

            if (c[i] > 32 && c[i] < 127)    //其他符号都转换为全角
                c[i] = (char) (c[i] + 65248);
        }
        return new String(c);
    }

    private static String deleteImgs(String content) {
        if (content!=null&&!TextUtils.isEmpty(content)){
            // 去掉所有html元素,
            String str = content.replaceAll("\\&[a-zA-Z]{1,10};", "").replaceAll(
                    "<[^>]*>", "");
            str = str.replaceAll("[(/>)<]", "");
            return str;
        }else {
            return "";
        }
    }

    public static String delete160(String des) {
        des = des.replaceAll("&#160;", "");
        des = des.replaceAll("&amp;#160;", "");
        des = des.replaceAll("\\s*", "");
        des = des.trim();
        return des;
    }

    //繁簡轉換
    public static String convertCC(String input) {
        int convertType = SpUtil.getIntValue(SHARED_READ_CONVERT_TYPE, 1);

        if (input.length() == 0)
            return "";

        return (convertType != 0) ? ZHConverter.getInstance(ZHConverter.TRADITIONAL).convert(input) : input;
    }

}

File: app/src/main/java/com/novel/read/utlis/StringUtils.kt
package com.novel.read.utlis

import android.text.TextUtils
import com.novel.read.widget.page.ReadSettingManager.Companion.SHARED_READ_CONVERT_TYPE

import com.spreada.utils.chinese.ZHConverter

object StringUtils {

    /**
     * 将文本中的半角字符，转换成全角字符
     */
    fun halfToFull(input: String): String {
        val text = deleteImgs(input)
        val c = text.toCharArray()
        for (i in c.indices) {
            if (c[i].toInt() == 32)
            //半角空格
            {
                c[i] = 12288.toChar()
                continue
            }
            //根据实际情况，过滤不需要转换的符号
            //if (c[i] == 46) //半角点号，不转换
            // continue;

            if (c[i].toInt() in 33..126)
            //其他符号都转换为全角
                c[i] = (c[i].toInt() + 65248).toChar()
        }
        return String(c)
    }

    private fun deleteImgs(content: String?): String {
        return if (content != null && !TextUtils.isEmpty(content)) {
            // 去掉所有html元素,
            var str =
                content.replace("&[a-zA-Z]{1,10};".toRegex(), "").replace("<[^>]*>".toRegex(), "")
            str = str.replace("[(/>)<]".toRegex(), "")
            str
        } else {
            ""
        }
    }

    fun delete160(des: String): String {
        var text = des
        text = text.replace("&#160;".toRegex(), "")
        text = text.replace("&amp;#160;".toRegex(), "")
        text = text.replace("\\s*".toRegex(), "")
        text = text.trim { it <= ' ' }
        return text
    }

    //繁簡轉換
    fun convertCC(input: String): String {
        val convertType = SpUtil.getIntValue(SHARED_READ_CONVERT_TYPE, 1)

        if (input.isEmpty())
            return ""

        return if (convertType != 0) ZHConverter.getInstance(ZHConverter.TRADITIONAL).convert(input) else input
    }

}
